src/features/post/model/hook/useQuery.ts

```diff
@@ -0,0 +1,48 @@
+import { postApis } from "../../../../entities/post/api";
+import { useMutation, useQuery, useSuspenseQuery } from "@tanstack/react-query";
+
+type ResPostsParams = {
+  skip: number;
+  limit: number;
+  searchQuery: string;
+  tag: string;
+};
+
+export const usePostsQuery = ({ skip, limit, searchQuery, tag }: ResPostsParams) => {
+  return useSuspenseQuery<ResPostsList>({
+    queryKey: ["posts", { skip, limit, searchQuery, tag }],
+    queryFn: () => {
+      if (searchQuery) {
+        return postApis.fetchSearchPosts(searchQuery);
+      } else if (tag !== "all" && tag !== "") {
+        return postApis.fetchPostsWithTag(tag);
+      }
+      return postApis.fetchPosts(skip, limit);
+    },
```

Comment:
p4;
else if를 사용하셨다면 가독성을 위해서 else로 마무리 작성하시는 것을 권장드립니다.

```suggestion
      if (searchQuery) {
        return postApis.fetchSearchPosts(searchQuery);
      } else if (tag !== "all" && tag !== "") {
        return postApis.fetchPostsWithTag(tag);
      } else {
        return postApis.fetchPosts(skip, limit);
      }
    },
```

Reply:
넵 감사합니다!